src/main/java/dev/binclub/javaception/runtime/ExecutionEngine.java

```diff
@@ -22,20 +32,696 @@ public static Object invokeMethodObj(InstanceOop instance, MethodInfo method, Ob
+				break;
+			case SWAP:
+				val = methodContext.pop();
+				val2 = methodContext.pop();
+				methodContext.push(val);
+				methodContext.push(val2);
+			case IADD:
+				methodContext.push((int) methodContext.pop() + (int) methodContext.pop());
+				break;
+			case LADD:
+				methodContext.push((long) methodContext.pop() + (long) methodContext.pop());
+				break;
+			case FADD:
+				methodContext.push((float) methodContext.pop() + (float) methodContext.pop());
+				break;
+			case DADD:
+				methodContext.push((double) methodContext.pop() + (double) methodContext.pop());
+				break;
+			case ISUB:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((int) val - (int) val2);
+				break;
+			case LSUB:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((long) val - (long) val2);
+				break;
+			case FSUB:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((float) val - (float) val2);
+				break;
+			case DSUB:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((double) val - (double) val2);
+				break;
+			case IMUL:
+				methodContext.push((int) methodContext.pop() * (int) methodContext.pop());
+				break;
+			case LMUL:
+				methodContext.push((long) methodContext.pop() * (long) methodContext.pop());
+				break;
+			case FMUL:
+				methodContext.push((float) methodContext.pop() * (float) methodContext.pop());
+				break;
+			case DMUL:
+				methodContext.push((double) methodContext.pop() * (double) methodContext.pop());
+				break;
+			case IDIV:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((int) val / (int) val2);
+				break;
+			case LDIV:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((long) val / (long) val2);
+				break;
+			case FDIV:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((float) val / (float) val2);
+				break;
+			case DDIV:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((double) val / (double) val2);
+				break;
+			case IREM:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((int) val % (int) val2);
+				break;
+			case LREM:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((long) val % (long) val2);
+				break;
+			case FREM:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((float) val % (float) val2);
+				break;
+			case DREM:
+				val2 = methodContext.pop();
+				val = methodContext.pop();
+				methodContext.push((double) val % (double) val2);
+				break;
+			case INEG:
+				methodContext.push((~(int) methodContext.pop()) + 1);
+			case FNEG:
+				Float fnum = (Float) methodContext.pop();
+				int fbits = Float.floatToIntBits(fnum);
+				int fneg = fbits ^ (1 << 31);
```

Comment:
Wait what? Can't you just do `methodContext.push(-((Float)methodContext.pop()));`?